### ISSUE TITLE
'Synced too far back' when booting a node and synchronizing

### ISSUE THREAD

**Issue by muXxer** [iotaledger/hornet#800]
Is the pruning correctly working?
People report about broken ledger states or "synced too far back" issues.

**Comment by luca-moser** [comment]
It actually also seems to happen without pruning being enabled.

Booting a node with following snapshot files:
```
> full snapshot, file ./full_snapshot.bin:
        - Snapshot time 2020-11-19 13:37:17 +0100 CET
        - Network ID 6530425480034647824
        - Ledger index 0
        - Snapshot index 0
        - UTXOs count 1
        - SEPs count 1
        - Milestone diffs count 0
> delta snapshot, file ./delta_snapshot.bin:
        - Snapshot time 2020-12-22 11:17:31 +0100 CET
        - Network ID 6530425480034647824
        - Ledger index 0
        - Snapshot index 113200
        - UTXOs count 0
        - SEPs count 200
        - Milestone diffs count 113200
```

results in:
https://ybin.me/p/d89cfdbb87d2eca3#1BRU13Wad+g8vB8Z679JnXGzhxrcDu/2OFepVewpmIk=

Used snapshot files:

[snapshot_files.zip](https://github.com/gohornet/hornet/files/5731426/snapshot_files.zip)

**Comment by muXxer** [comment]
Fixed with #806
